packages/openid4vc-client/src/OpenId4VcClientService.ts

```diff
@@ -1,36 +1,65 @@
-import type { AgentContext, W3cCredentialRecord } from '@aries-framework/core'
-import type { EndpointMetadata, Jwt } from '@sphereon/openid4vci-client'
+import type { AgentContext } from '@aries-framework/core'
+import type { AccessTokenResponse, EndpointMetadata, Jwt } from '@sphereon/openid4vci-client'
 
 import {
+  AriesFrameworkError,
+  DidsApi,
   getKeyFromVerificationMethod,
+  Hasher,
   inject,
+  injectable,
   InjectionSymbols,
   isJwtAlgorithm,
-  Logger,
-  DidsApi,
-  AriesFrameworkError,
-  injectable,
   JsonEncoder,
   JsonTransformer,
-  W3cCredentialService,
-  W3cVerifiableCredential,
   JwsService,
   jwtKeyAlgMapping,
+  Logger,
+  TypedArrayEncoder,
+  W3cCredentialService,
+  W3cVerifiableCredential,
 } from '@aries-framework/core'
 import {
   Alg,
   AuthzFlowType,
+  CodeChallengeMethod,
   CredentialRequestClientBuilder,
   OpenID4VCIClient,
   ProofOfPossessionBuilder,
 } from '@sphereon/openid4vci-client'
+import { randomStringForEntropy } from '@stablelib/random'
 
-export interface PreAuthorizedOptions {
+export interface PreAuthCodeFlowOptions {
   issuerUri: string
   kid: string
   checkRevocationState: boolean
 }
 
+export interface AuthCodeFlowOptions extends PreAuthCodeFlowOptions {
+  clientId: string
+  authorizationCode: string
+  codeVerifier: string
+  redirectUri: string
+}
+
+export enum AuthFlowType {
+  AUTHORIZATION_CODE_FLOW = 'Authorization Code Flow',
```

Comment:
Nit: but we usually use pascal case for enum keys and values, and don't use spaces etc in the values. 

Would be good to keep this consistent with AFJ more than with the sphereon lib